dm/pkg/checker/utils.go

```diff
@@ -136,7 +138,9 @@ func markCheckError(result *Result, err error) {
 		} else {
 			state = StateFailure
 		}
-		result.State = state
+		if result.State != StateFailure {
```

Comment:
could please explian why update state only when state not equal

Reply:
`markCheckError` may be called repeatedly. So `StateFailure` can be covered by `StartWarn`

Reply:
could you add a comment aroud this code?